code/modules/overmap/ships/computers/comms.dm

```diff
@@ -0,0 +1,245 @@
+
+	for(var/obj/machinery/shipcomms/comms in linked.comms_masers)
+		var/functional = (comms.stat & (BROKEN|NOPOWER))
+		broadcasters += list(list(
+			"name" = comms.name, 
+			"coords" = "[comms.x],[comms.y],[comms.z]", 
+			"desc" = (functional ? "inactive" : "active")
+		))
+		if(functional)
+			for(var/obj/effect/overmap/entity in comms.get_nearby_entitities())
+				if(entity in printing_entities)
+					continue
+				printing_entities += entity
+				var/has_receiver = FALSE
+				for(var/obj/machinery/shipcomms/receiver/antenna in entity.comms_antennae)
+					if(!(antenna.stat & (BROKEN|NOPOWER)))
+						has_receiver = TRUE
+				if(!has_receiver)
+					continue
+				var/has_broadcaster = FALSE
+				for(var/obj/machinery/shipcomms/broadcaster/maser in entity.comms_masers)
+					if(!(maser.stat & (BROKEN|NOPOWER)))
+						has_broadcaster = TRUE
+				entities += list(list(
+					"name" = entity.name, 
+					"coords" = "[entity.x],[entity.y]", 
+					"desc" = entity.desc, 
+					"status" = ((has_broadcaster && has_receiver) ? "Responding to comms pings." : "Not responding to comms pings.")
+				))
+
+	for(var/obj/machinery/shipcomms/comms in linked.comms_antennae)
+		receivers += list(list(
+			"name" = comms.name, 
+			"coords" = "[comms.x],[comms.y],[comms.z]", 
+			"desc" = ((comms.stat & (BROKEN|NOPOWER)) ? "inactive" : "active")
+		))
+
+	if(!length(entities))
+		entities += list(list(
+			"name" = "None.", 
+			"coords" = "-,-", 
+			"desc" = "-",
+			"status" = "No local entities responding to comms ping."
+		))
+	if(!length(broadcasters))
+		broadcasters += list(list(
+			"name" = "None.", 
+			"coords" = "-,-,-", 
+			"desc" = "No broadcasters available."
+		))
+	if(!length(receivers))
+		receivers += list(list(
+			"name" = "None.", 
+			"coords" = "-,-,-", 
+			"desc" = "No receivers available."
+		))
+
+	data["allow_ident_change"] = linked.can_switch_ident
+	data["ident_enabled"] =      linked.ident_transmitter
+
+	data["entities"] =     entities
+	data["broadcasters"] = broadcasters
+	data["receivers"] =    receivers
+
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "shipcomms.tmpl", "[linked.name] Communication Screen", 380, 530)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/machinery/computer/ship/comms/OnTopic(var/mob/user, var/list/href_list)
+	if(..())
+		return TOPIC_HANDLED
+	if (!linked)
+		return TOPIC_NOACTION
+	if(href_list["toggle_comms_visibility"])
+		if(linked.can_switch_ident)
+			linked.ident_transmitter = !linked.ident_transmitter
+			. = TOPIC_REFRESH
+
+/obj/machinery/shipcomms
+	density = TRUE
+	anchored = TRUE
+	idle_power_usage = 1000
+	req_access = list(access_tcomsat)
+	icon = 'icons/obj/machines/tcomms/comms.dmi'
+	var/effective_range = 1 // TODO: upgrade with components.
+	var/enabled = TRUE
+
+/obj/machinery/shipcomms/proc/toggle()
+	if(use_power && !(stat & BROKEN))
+		update_use_power(!use_power)
```

Comment:
The `if` in the line above suggests that once you toggle it off, you can't toggle it on ever again.